File: Python/Heap_PriorityQueue_Medium_KthLargestElementInAnArray_215.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        heap = []
        heapLimit = k

        for num in nums:
            heapq.heappush(heap, num)

            if len(heap) > heapLimit:
                heapq.heappop(heap)

        return heap[0]
        

File: C++/Heap_PriorityQueue_Medium_KthLargestElementInAnArray_215.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) 
    {
        priority_queue<int, vector<int>, greater<int>> pq;
        int heapLimit = k;

        for(int i = 0; i < nums.size(); i++)
        {
            pq.push(nums[i]);

            if(pq.size() > heapLimit)
            {
                pq.pop();
            }
        }

        return pq.top();
    }
};